searx/engines/openalex.py

```diff
@@ -0,0 +1,210 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""OpenAlex (Scientific works)
+
+References:
+- API overview: https://docs.openalex.org/how-to-use-the-api/api-overview
+
+The engine queries the OpenAlex Works endpoint and maps fields to the
+`paper.html` result template.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+from urllib.parse import urlencode
+
+# about
+about = {
+    "website": "https://openalex.org/",
+    "wikidata_id": "Q110718454",
+    "official_api_documentation": "https://docs.openalex.org/how-to-use-the-api/api-overview",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+
+# engine dependent config
+categories = ["science", "scientific publications"]
+paging = True
+search_url = "https://api.openalex.org/works"
+
+# Optional: include your email for OpenAlex polite pool. Can be set from settings.yml
+# engines: - name: openalex; engine: openalex; mailto: "[email protected]"
+
```

Comment:
```suggestion
mailto = ""
```
Please declare mailto here instead of accessing it via `globals()`.

Reply:
To avoid name collisions: better use `openalex_mailto`.